File: src/ReadStorm.Application/Abstractions/IBookshelfUseCase.cs
using ReadStorm.Domain.Models;

namespace ReadStorm.Application.Abstractions;

public interface IBookshelfUseCase
{
    Task<IReadOnlyList<BookRecord>> GetAllAsync(CancellationToken cancellationToken = default);

    Task AddAsync(BookRecord book, CancellationToken cancellationToken = default);

    Task RemoveAsync(Guid bookId, CancellationToken cancellationToken = default);

    Task UpdateProgressAsync(Guid bookId, ReadingProgress progress, CancellationToken cancellationToken = default);
}

File: pureopus/app/src/main/java/com/readstorm/app/application/abstractions/IBookshelfUseCase.kt
package com.readstorm.app.application.abstractions

import com.readstorm.app.domain.models.BookRecord
import java.util.UUID

interface IBookshelfUseCase {
    suspend fun getAll(): List<BookRecord>

    suspend fun add(book: BookRecord)

    suspend fun remove(bookId: UUID)

    suspend fun updateProgress(bookId: UUID, progress: BookRecord.ReadingProgress)
}
